Site Setting order when Containment flag is on

### Description

When `Settings Containment` flag is turned on, enter Site Settings causes debug version to fail on assertion
```
    java.lang.AssertionError: Remove the order adjustment if the prefs are removed from upstream
    	at org.chromium.components.browser_ui.site_settings.BraveSiteSettingsPreferencesBase.configureBravePreferences(BraveSiteSettingsPreferencesBase.java:88)
    	at org.chromium.components.browser_ui.site_settings.BraveSiteSettingsPreferencesBase.onCreate(BraveSiteSettingsPreferencesBase.java:36)

```
and on Release version the `Automatically remove permission` section is not at the end

Flag|No containments|With containments
--|--|--
<img width="1080" height="2400" alt="Image" src="https://github.com/user-attachments/assets/15a3d2c5-88a2-4e8d-9da1-03963ff24a9a" />|<img width="1080" height="2400" alt="Image" src="https://github.com/user-attachments/assets/d263bf7d-20ef-4af8-a569-971d6c148a5c" />|<img width="1080" height="2400" alt="Image" src="https://github.com/user-attachments/assets/be85e252-12ee-4b4e-a6a2-f28417b4c4c8" />



### Steps to reproduce

1. Enable, `Android Settings Containment` flag, restart
2. Go to Settings => Site Settings
3. Scroll down

### Actual result

`Automatically remove permission` is not at the very end

### Expected result

`Automatically remove permission` should be at the very end

### Reproduces how often

Easily reproduced

### Brave version

v1.91.49

### Device

Any

### Channel information

- [ ] release (stable)
- [ ] beta
- [x] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

### Miscellaneous information

_No response_

--- a/android/javatests/org/chromium/chrome/browser/site_settings/BraveSiteSettingsTest.java
+++ b/android/javatests/org/chromium/chrome/browser/site_settings/BraveSiteSettingsTest.java
@@ -14,6 +14,9 @@ import org.junit.runner.RunWith;
 
 import org.chromium.base.test.util.Batch;
 import org.chromium.base.test.util.Feature;
+import org.chromium.base.test.util.Features.DisableFeatures;
+import org.chromium.base.test.util.Features.EnableFeatures;
+import org.chromium.chrome.browser.flags.ChromeFeatureList;
 import org.chromium.chrome.browser.settings.SettingsActivity;
 import org.chromium.chrome.test.ChromeJUnit4ClassRunner;
 import org.chromium.components.browser_ui.site_settings.SiteSettings;
@@ -32,6 +35,7 @@ public class BraveSiteSettingsTest {
     @Test
     @SmallTest
     @Feature({"Preferences"})
+    @DisableFeatures(ChromeFeatureList.ANDROID_SETTINGS_CONTAINMENT)
     public void testSiteSettingsMenuPermissionAutorevocationOrder() {
         final SettingsActivity settingsActivity = SiteSettingsTestUtils.startSiteSettingsMenu("");
         SiteSettings siteSettings = (SiteSettings) settingsActivity.getMainFragment();
@@ -47,6 +51,26 @@ public class BraveSiteSettingsTest {
         Assert.assertEquals(prefSolanaConnectedSites.getOrder(), prefDivider.getOrder() - 1);
         Assert.assertEquals(
                 prefSolanaConnectedSites.getOrder(), prefPermissionAutorevocation.getOrder() - 2);
+        settingsActivity.finish();
+    }
+
+    /** The same but with `Android Settings Containment` flag enabled */
+    @Test
+    @SmallTest
+    @Feature({"Preferences"})
+    @EnableFeatures(ChromeFeatureList.ANDROID_SETTINGS_CONTAINMENT)
+    public void testSiteSettingsMenuPermissionAutorevocationOrderWithContainment() {
+        final SettingsActivity settingsActivity = SiteSettingsTestUtils.startSiteSettingsMenu("");
+        SiteSettings siteSettings = (SiteSettings) settingsActivity.getMainFragment();
+        Assert.assertNotNull(siteSettings);
+        Preference prefPermissionAutorevocation =
+                siteSettings.findPreference(PERMISSION_AUTOREVOCATION_KEY);
+        Assert.assertNotNull(prefPermissionAutorevocation);
+        Preference prefSolanaConnectedSites =
+                siteSettings.findPreference(SOLANA_CONNECTED_SITES_KEY);
+        Assert.assertNotNull(prefSolanaConnectedSites);
+        Assert.assertEquals(
+                prefSolanaConnectedSites.getOrder(), prefPermissionAutorevocation.getOrder() - 1);
         settingsActivity.finish();
     }
 }

--- a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BraveSiteSettingsPreferencesBase.java
+++ b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BraveSiteSettingsPreferencesBase.java
@@ -82,19 +82,24 @@ public class BraveSiteSettingsPreferencesBase extends BaseSiteSettingsFragment {
         // We want to place these Settings at the bottom.
         // See https://github.com/brave/brave-browser/issues/46547
         // for the context
-        Preference prefDivider = getPreferenceScreen().findPreference(DIVIDER_KEY);
         Preference prefPermissionAutorevocation =
                 getPreferenceScreen().findPreference(PERMISSION_AUTOREVOCATION_KEY);
-        assert prefDivider != null && prefPermissionAutorevocation != null
+        assert prefPermissionAutorevocation != null
                 : "Remove the order adjustment if the prefs are removed from upstream";
-        if (prefDivider != null && prefPermissionAutorevocation != null) {
+        if (prefPermissionAutorevocation != null) {
             Preference prefSolanaConnectedSites =
                     getPreferenceScreen().findPreference(SOLANA_CONNECTED_SITES_KEY);
             // Solana preference may be removed if wallet is disabled by policy.
             if (prefSolanaConnectedSites != null) {
                 int solanaConnectedSitesOrder = prefSolanaConnectedSites.getOrder();
-                prefDivider.setOrder(solanaConnectedSitesOrder + 1);
-                prefPermissionAutorevocation.setOrder(solanaConnectedSitesOrder + 2);
+                Preference prefDivider = getPreferenceScreen().findPreference(DIVIDER_KEY);
+                if (prefDivider == null) {
+                    // There is divider when `Android Settings Containment` flag is turned on
+                    prefPermissionAutorevocation.setOrder(solanaConnectedSitesOrder + 1);
+                } else {
+                    prefDivider.setOrder(solanaConnectedSitesOrder + 1);
+                    prefPermissionAutorevocation.setOrder(solanaConnectedSitesOrder + 2);
+                }
             }
         }
     }

